.github/workflows/docker-base-images.yml

```diff
@@ -0,0 +1,74 @@
+name: "Docker base images"
+on:
+  # This GitHub Action will be invoked automatically from the main Azure DevOps build
+  # This job will only be launched when we are building the main branch (a snapshot) or when we are generating a release.
+  workflow_dispatch:
+    inputs:
+      azdo_build_id:
+        description: 'The specific AzDo build from which the release artifacts will be downloaded.'
+        required: true
+      is_snapshot:
+        description: 'True when we are building main branch, false when we are building a release'
+        required: true
+jobs:
+  build-and-publish-base-image:
+    runs-on: ubuntu-latest
+    env:
+      AZURE_DEVOPS_TOKEN: "${{ secrets.AZURE_DEVOPS_TOKEN }}"
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+
+    - uses: actions/setup-dotnet@v1
+      with:
+        dotnet-version: '7.0.101'
+
+    - name: "Get current version"
+      id: versions
+      run: ./tracer/build.sh OutputCurrentVersionToGitHub
+
+    - name: "Download build assets from Azure Pipelines"
+      id: assets
+      run: ./tracer/build.sh DownloadAzurePipelineFromBuild 
+      env:
+        AzureDevopsBuildId: "${{ github.event.inputs.azdo_build_id }}"
+
+    - name: Copy tooling files to artifacts path
+      shell: bash
+      run: |
+        echo "OUTPUT ARTIFACT PATH: ${{steps.assets.outputs.artifacts_path}}"
+        cp ./tracer/build/_build/docker/reliability-env/* ${{steps.assets.outputs.artifacts_path}}
+
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v2
+
+    - name: Set up Docker Buildx
+      id: buildx
+      uses: docker/setup-buildx-action@v2
+
+    - name: Generate image tags
+      id: docker-base-image-tags
+      shell: bash
+      run: |
+        if [ "$is_snapshot" = "true" ]; then
+          echo "tag-names=ghcr.io/datadog/dd-trace-dotnet/dd-trace-dotnet:latest_snapshot" >> $GITHUB_OUTPUT
+        else
+          echo "tag-names=ghcr.io/datadog/dd-trace-dotnet/dd-trace-dotnet:latest" >> $GITHUB_OUTPUT
```

Comment:
This case will almost never happen (unless started manually). 
The trigger in the AzDo pipeline happens on merge to master or to a release branch (like hotfixes). As we release from master in 99% of cases, you will always have a `latest_snapshot` tag. Is is what you want?

Reply:
I would like to generate the latest_snapshot for every commit on master.
I thought we will make a commit on refs/heads/release/ when we generate release (is_snapshot=false)

Reply:
Sorry for the delay. The changes look good, except we don't always commit to a `refs/heads/release/` branch when we make a new release (like the one we just published today). We usually take the latest commit from master and annotate it with a git tag. If you need a `latest` image, maybe we can invoke this GitHub Action from our `create_draft_release.yml` GitHub Action which starts the automated release process. What do you think?